lib/pure/os.nim

```diff
@@ -672,7 +672,7 @@ template walkCommon(pattern: string, filter) =
           if dotPos < 0 or idx >= ff.len or ff[idx] == '.' or
               pattern[dotPos+1] == '*':
             yield splitFile(pattern).dir / extractFilename(ff)
-        if findNextFile(res, f) == 0'i32: break
+        if findNextFile(res, f) == 0'i32 and getLastError() == 18: break # ERROR_NO_MORE_FILES=18
```

Comment:
Add this constant to ``winlean``.

Reply:
If it's a different error it should be raised. Use ``raiseOsError()`` for that.

Reply:
if `raiseOsError()`, then the finding process will be stopped and has no change to continue, and `walkDir[Rec]` will never have the chance to walk the whole directory

Reply:
yes, that's what should happen. Errors shouldn't be ignored, unless they can be handled by this code.